### PR TITLE
Use webpack-merge to make a prod webpack config

### PR DESCRIPTION
**Pull request by thallada**
@edx/educator-dahlia

### CODE REVIEW

File: config/webpack.common.config.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  ],
+  output: {
+    filename: 'bundle.js',
+    path: path.resolve(__dirname, '../dist'),
+  },
+  resolve: {
+    extensions: ['.js', '.jsx'],
+  },
+  plugins: [
+    new HtmlWebpackPlugin({
```
Comment (arizzitano) [inline config/webpack.common.config.js]: We may not want to use this plugin in production if we are just outputting js files for studio's html to consume.

File: config/webpack.prod.config.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+const Merge = require('webpack-merge');
+const commonConfig = require('./webpack.common.config.js');
+const path = require('path');
+const webpack = require('webpack');
+
+module.exports = Merge.smart(commonConfig, {
+  devtool: 'source-map',
+  module: {
+    rules: [
+      {
```
Comment (arizzitano) [inline config/webpack.prod.config.js]: I think we need this rule in the common config, no?

Comment (arizzitano) [reply config/webpack.prod.config.js]: oops it's also in the dev config nvm

File: config/webpack.common.config.js (hunk 1 above)
Comment (thallada) [reply config/webpack.common.config.js]: Ah, good point. Also, what do you think about naming the output bundle `studio-frontend.js` in prod?

Comment (arizzitano) [reply config/webpack.common.config.js]: sounds good to me!

**Pull request merged**